src/nrnmpi/nrnmpi.cpp

```diff
@@ -111,17 +94,29 @@ for (i=0; i < *pargc; ++i) {
 
         /* only call MPI_Init if not already initialized */
         if (!flag) {
+            // occasional problems with openmpi when mpi requested but
+            // launched without an mpiexec or equivalent. If some MPI
+            // implementation makes nontrivial use of argv, then perhaps
+            // restrict this hack to openmpi
+            static int argc = 1;
+            static char** argv;
+            if (!argv) {
+                argv = (char**)malloc(sizeof(char*));
```

Comment:
can't we use a static array size 1 and not have the malloc? `static char* argv[1];`

Reply:
I was confused about how to do that since the signature is
```
int MPI_Init(int *argc, char ***argv)
```
So changing to
```
            static int argc = 1;
            static char* argv[1]; 
            if (!argv[0]) {
                argv[0] = strdup(*pargc ? (*pargv)[0] : "nrniv");
            }
```
gives:
```
 ../src/nrnmpi/nrnmpi.cpp: In function ‘void nrnmpi_init(int, int*, char***)’:
../src/nrnmpi/nrnmpi.cpp:110:67: error: cannot convert ‘char* (*)[1]’ to ‘char***’ for argument ‘2’ to ‘int MPI_Init_thread(int*, char***, int, int*)’
```

Beside the point but, in the description it claims
> Open MPI accepts the C/C++ argc and argv arguments to main, but neither modifies, interprets, nor distributes them: